cpp/src/transform/jit/kernel.cu

```diff
@@ -121,6 +165,15 @@ CUDF_KERNEL void kernel(cudf::mutable_column_device_view_core const* outputs,
   thread_index_type const size   = outputs[0].size();
 
   for (auto i = start; i < size; i += stride) {
+    if constexpr (has_nulls) {
+      if ((false || ... || In::is_null(inputs, i))) {
+        Out::set_null(outputs, i);
+        continue;
+      } else {
+        Out::set_valid(outputs, i);
+      }
+    }
```

Comment:
I think this would be less efficient than just gathering the bitmasks separately and applying it to the output column. Also, I think this will be a race condition since multiple threads may be trying to update the same bytes.

Reply:
> I think this would be less efficient than just gathering the bitmasks separately and applying it to the output column

Ok, I'll do that.

> Also, I think this will be a race condition since multiple threads may be trying to update the same bytes.

It's an atomic bit operation

Reply:
Also, we have found there is no significant performance improvement for making `has_nulls` a template argument where the code divergence is very short -- like in this case where the bit is just checked. I'd rather the `has_nulls` be a runtime parameter or just check for the validity-mask all the time. It is not worth (compiler) generating separating kernels for just the single-bit check.
Perhaps some benchmarks that prove this worthy of a template parameter?